src/Analyzer/Resolve/ScopeAliases.h

```diff
@@ -17,6 +17,9 @@ struct ScopeAliases
     /// Alias name to table expression node
     std::unordered_map<std::string, QueryTreeNodePtr> alias_name_to_table_expression_node;
 
+    /// Cached fully resolved expression aliases
+    std::unordered_map<std::string, QueryTreeNodePtr> alias_name_to_resolved_expression_node;
```

Comment:
The hash table key must have `IdentifierLookup` type, because the result depends on the lookup context.